### PR TITLE
Flows unit tests

### PR DESCRIPTION
**Pull request by istalker2**
This change is [Reviewable](https://reviewable.io/reviews/mirantis/k8s-appcontroller/238)

### CODE REVIEW

File: pkg/mocks/client.go [hunk 1]
```diff
@@ -55,6 +64,22 @@ func NewClient(objects ...runtime.Object) client.Interface {
...
 
+func NewClientWithFake(objects ...runtime.Object) (client.Interface, *testing.Fake) {
+	return newClientWithFake(makeVersionsList(v1beta1.SchemeGroupVersion), objects...)
+}
+
 func NewClient1_4(objects ...runtime.Object) client.Interface {
 	return newClient(makeVersionsList(v1alpha1.SchemeGroupVersion), objects...)
 }
+
+func normalizeName(name string) string {
```
Comment (nebril) [inline pkg/mocks/client.go]: Why not use the func from cmd/format/format.go?

File: pkg/mocks/resdefs.go [hunk 2]
```diff
@@ -24,10 +24,13 @@ import (
 )
 
 func MakeResourceDefinition(name string) *client.ResourceDefinition {
+	resName := normalizeName(name)
```
Comment (nebril) [inline pkg/mocks/resdefs.go]: Should use format.NormalizeName

File: pkg/scheduler/dependency_graph_test.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+	"github.com/Mirantis/k8s-AppController/pkg/interfaces"
+	"github.com/Mirantis/k8s-AppController/pkg/mocks"
+)
+
+// TestAllocateReplicas tests replica allocation in different configurations
+func TestAllocateReplicas(t *testing.T) {
+	flow := mocks.MakeFlow("flow").Flow
+	c := mocks.NewClient()
+	sched := New(c, nil, 0).(*Scheduler)
+	newReplicas1, deleteReplicas, _ := sched.allocateReplicas(flow, interfaces.DependencyGraphOptions{ReplicaCount: 3})
```
Comment (nebril) [inline pkg/scheduler/dependency_graph_test.go]: would be nice to check that err (_ here) is not nil in at least one call

File: cmd/run_test.go [hunk 4]
```diff
@@ -56,3 +56,44 @@ func TestLabelFlag(t *testing.T) {
...
+		{"  x1   y2  ", "x1", "y2", false},
+		{"a=b=c", "a", "b=c", false},
+		{"a", "", "", true},
+		{"", "", "", true},
+		{"=", "", "", true},
+		{"a=", "a", "", false},
+		{"=a", "", "", true},
+	}
+
+	for _, x := range table {
```
Comment (istalker2) [inline cmd/run_test.go]: I followed advice from here: https://github.com/golang/go/wiki/TableDrivenTests

File: pkg/mocks/client.go (hunk 1 above)
Comment (istalker2) [inline pkg/mocks/client.go]: it is formatNormalize name (not exported) and doesn't handle "/" symbol (and should not do it). Considering that this is a very simple method of only few lines of code it seems that having separate methods is a more natural solution

File: pkg/mocks/resdefs.go (hunk 2 above)
Comment (istalker2) [inline pkg/mocks/resdefs.go]: see above

File: pkg/scheduler/dependency_graph_test.go (hunk 3 above)
Comment (istalker2) [inline pkg/scheduler/dependency_graph_test.go]: ack, fixed

**Comment by istalker2** [comment]
PTAL

**Pull request merged**